src/main/java/com/pnu/dev/shpaltaif/service/UserServiceImpl.java

```diff
@@ -75,16 +77,28 @@ public void create(CreateUserDto createUserDto) {
         if (userRepository.existsByUsername(createUserDto.getUsername())) {
             throw new ServiceException("Логін уже використовується!");
         }
+        if (createUserDto.getRole().equals(UserRole.ROLE_WRITER)) {
+            if (isBlank(createUserDto.getName())) {
+                throw new ServiceException("Ім'я повинно бути вказаним");
+            }
+            if (isBlank(createUserDto.getSurname())) {
+                throw new ServiceException("Прізвище повинно бути вказаним");
+            }
+        }
 
         User user = User.builder()
                 .username(createUserDto.getUsername())
                 .password(bCryptPasswordEncoder.encode(createUserDto.getPassword()))
-                .role(UserRole.ROLE_WRITER)
+                .role(createUserDto.getRole())
                 .active(Boolean.TRUE)
                 .build();
 
         User savedUser = userRepository.save(user);
 
+        if (savedUser.getRole().equals(UserRole.ROLE_EDITOR)) {
```

Comment:
This is the case when such approach cannot be used...
Wrap creation of public account in "if" block and check for user role.
Reason - the code have to reflect expected behaviour of the system, is such case it looks like we expect that only editor will not have a public accout, but it reality we expect only writer to have it.

Reply:
You can compare enum via `==` operator. It works a little bit faster.